src/shader/shaderUtils.js

```diff
@@ -225,4 +225,8 @@ export const shaderUtils = /* glsl */`
 		return vec3( sinPhi * cos( theta ), cos( phi ), sinPhi * sin( theta ) );
 
 	}
+
+	float acosApprox(float x) {
+		return ( -0.69813170079773212 * x * x - 0.87266462599716477 ) * x + 1.5707963267948966;
+	}
```

Comment:
Is there a reference for this approximation? Can we add a comment with some information on it?

Reply:
> Great work, this is excellent! Thank you - I hadn't thought the built in acos would cause these kinds of issues. Do you know why that is? Is it just known to be an imprecise call?

`acos` on the GPU is implemented with something like this https://developer.download.nvidia.com/cg/acos.html - the number of operations and the use of `sqrt` all contribute to numerical precision issues, this is further worsened when denormals are flushed to zero. 

The reported absolute error <= 6.7e-5 detailed in the above page should be fine in most cases but when interacting with other functions in this iterative setting this gives rise to pathological fixed-points.

Reply:
> Is there a reference for this approximation? Can we add a comment with some information on it?

This is a Lagrange interpolating polynomial for the points x∈{-1, -½, 0, ½, 1} e.g. https://mathworld.wolfram.com/LagrangeInterpolatingPolynomial.html 

The values themselves come from https://stackoverflow.com/questions/3380628/fast-arc-cos-algorithm/answer#3380723

The cubic polynomial factored as it is saves a few operations, which is a small extra win.

Reply:
I'll add a comment and update the PR.